src/main/java/seedu/duke/command/EditCommand.java

```diff
@@ -126,15 +138,85 @@ public void setDate(LocalDate date) {
      * @param storage      An instance of the Storage class.
      */
     @Override
-    public void execute(TransactionList transactions, Ui ui, Storage storage) {
-        // Dummy output for test
-        System.out.println(String.format("Entry number: %d\nType: %s\nDesc: %s\n$: %d\nCat: %s, Date: %s",
-                entryNumber,
-                type,
-                description,
-                amount,
-                category,
-                date.toString()));
+    public void execute(TransactionList transactions, Ui ui, Storage storage) throws MoolahException {
+        try {
+            editLogger.setLevel(Level.SEVERE);
+            editLogger.log(Level.INFO, "Edit Command checks whether the index is valid "
+                    + "before executing the command.");
+            int index = entryNumber;
+            Transaction entry = transactions.getEntry(index - 1);
+            boolean isInputValid = true;
+            int numberOfTransactions = transactions.size();
+
+            if ((index > numberOfTransactions) || (index <= 0)) {
+                isInputValid = false;
+            }
+            assert index > 0;
+
+            if (isInputValid) {
+                String newType = type;
+                String newDescription = description;
+                int newAmount = amount;
+                LocalDate newDate = date;
+                String newCategory = category;
+
+                if (newType == null) {
+                    newType = entry.getType();
+                }
+
+                if (newType.equals("expense")) {
+                    if (newDate == null) {
```

Comment:
May wanna extract the arrowhead code out and possibly put into a different method when you revisit this code. Code in this method is a bit long .